app/controllers/items_controller.rb

```diff
@@ -624,23 +719,22 @@ def save_and_reindex
     reindex @object unless params[:bulk]
   end
 
-  # check that the user can carry out this item modification
-  def forbid_modify
-    return true if current_user.is_admin || @object.can_manage_content?(current_user.roles @apo)
-    render :status => :forbidden, :text => 'forbidden'
-    false
+  # ---
+  # Dor::Workflow utils
+
+  def dor_accession_error?(object, task)
+    dor_accession_status(object, task) == 'error'
   end
 
-  def forbid_view
-    return true if current_user.is_admin || @object.can_view_content?(current_user.roles @apo)
-    render :status => :forbidden, :text => 'forbidden'
-    false
+  def dor_accession_status(object, task)
+    Dor::WorkflowService.get_workflow_status('dor', object.pid, 'accessionWF', task)
   end
 
-  def enforce_versioning
-    # if this object has been submitted, doesnt have an open version, and isnt sitting at sdr-ingest with a hold, they cannot change it.
-    return true if @object.allows_modification?
-    render status: :forbidden, text: 'Object cannot be modified in its current state.'
-    false
+  def dor_lifecycle(object, stage)
+    Dor::WorkflowService.get_lifecycle('dor', object.pid, stage)
+  end
+
+  def set_dor_accession_status(object, task, status)
+    Dor::WorkflowService.update_workflow_status('dor', object.pid, 'accessionWF', task, status)
   end
 end
```

Comment:
Various utils to simplify the calls to `Dor::WorkflowService` and shorten some long lines in this controller.